src/_C004.ino

```diff
@@ -124,7 +124,7 @@ bool do_process_c004_delay_queue(int controller_number, const C004_queue_element
   String postStr = do_create_http_request(
     hostName, F("POST"),
     F("/update"), // uri
-    F(""),           // auth_header
+    EMPTY_STRING,           // auth_header
```

Comment:
Is this a generic improvement, simply replace all `F("")` by `EMPTY_STRING` ? Or does it require a `__FlashStringHelper` reference?

(If so then I'll start updating that in my work in progress 😉)

Reply:
Nope, it is slightly more complex :)

For those functions that return a `String` object, this is an optimization as you don't need to create a String object, but simply reference a static const existing one.
But it is a bigger reduction in size if you can create functions that return a flash string instead of having code for every assignment of a flash string to a `String` object.

e.g. a typical `toString` function:
```c++
String toString(myEnumClassType enumValue) {
  switch (enumValue){
    case myEnumClassType::bla:
      return F("bla");
  }
}
```
This return will effectively generate `return String(F("bla"));` for every flash string return calls.
By returning a flash string type, you only cast it to `String` after the function is called.  (or not at all if it can be kept as a flash string)
For example the webserver markup function calls that now also accept a flash string array, which only holds a pointer instead of allocating memory for the string only to dump it to the TXBuffer object, which also can handle flash strings without conversion.

That's the true change of this PR.